akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -60,6 +60,15 @@ object Flow {
       case other            => new Flow(scaladsl.Flow.fromGraph(other))
     }
 
+  /**
+   * Defers the creation of a [[Flow]] until materialization. The `factory` function
+   * exposes [[ActorMaterializer]] which is going to be used during materialization and
+   * [[Attributes]] of the [[Flow]] returned by this method.
+   */
+  def setup[I, O, M](
+      factory: BiFunction[ActorMaterializer, Attributes, Flow[I, O, M]]): Flow[I, O, CompletionStage[M]] =
```

Comment:
why `ActorMaterializer` instead of `Materializer`?

Reply:
We had a usecase where in the `setup` we want to access the `ActorSystem` to get to the Akka Extension.

Reply:
but is there anything in this implementation that requires `ActorMaterializer`? for consistency with other APIs (e.g. `GraphStageLogic.materializer`) I think we should continue using `Materializer` as the type. In practice it's only `ActorMaterializer` and in your usage in Alpakka you can cast it to an `ActorMaterializer`.

Reply:
In Alpakka we need the `ActorMaterializer` to get the `ActorSystem` out of it for Akka Http, for example. An alternative could be to expose all three, ActorSystem, Materializer and Attributes.

Reply:
Exposing ActorSystem as a separate parameter is no better than having `ActorMaterializer`.
Seems it is important for you to not cast in Alpakka so I accept.